bee-message/src/message.rs

```diff
@@ -17,8 +17,7 @@ pub const MESSAGE_LENGTH_MAX: usize = 32768;
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Message {
     network_id: u64,
-    parent1: MessageId,
-    parent2: MessageId,
+    parents: Vec<MessageId>,
     payload: Option<Payload>,
     nonce: u64,
 }
```

Comment:
also more of a comment to think about, then a merge blocker:

A `Message` has to have `1..` parents, not `0..`. Yes, we have guards in place that might always be enough to guarantee, that the `Vec` is nevaaa evaaa empty, but I'ld like to propose this type, that tells the compiler that we truely mean the former than the latter:

```rust
struct NonEmptyVec {
    parent1: MessageId,
    Vec<MessageId>,
} 
```
As an additional benefit: for Messages with only 1 parent there is no extra allocation. And then obviously we'ld implement Iterator on it and some API similar to Vec. I think it is easy to implement and will make the node more robost agains programmer errors.

Reply:
A simpler alternative is to make `NonEmptyParents` a simple wrapper of `Vec`, that ensures it is always initialized with at least one parent. This may be simpler, has the same guarantee, but will always allocate.